### PR TITLE
Block Library: Fix React does not recognize isSelected prop in ComplementaryAreaToggle.

### PR DESCRIPTION
**Pull request by torounit**
## Description

fix error `Warning: React does not recognize the `isSelected` prop on a DOM element.`

```
Warning: React does not recognize the `isSelected` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isselected` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in button (created by ForwardRef(Button))
    in Tooltip (created by ForwardRef(Button))
    in ForwardRef(Button) (created by ComplementaryAreaToggle)
    in ComplementaryAreaToggle (created by Context.Consumer)
...
```

## Types of changes

Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/interface/src/components/complementary-area-toggle/index.js [hunk 1]
```diff
@@ -32,18 +32,21 @@ function ComplementaryAreaToggle( {
...
 		'core/interface'
 	);
+	const componentProps = { ...props };
+	if ( ComponentToUse !== Button ) {
+		componentProps.isSelected = isSelected;
+	}
 	return (
 		<ComponentToUse
 			icon={ selectedIcon && isSelected ? selectedIcon : icon }
-			isSelected={ isSelected }
```
Comment (youknowriad) [inline packages/interface/src/components/complementary-area-toggle/index.js:38]: Maybe we should just remove it? Is it useful anywhere?

If I understand properly the components that can be used as `as` should follow the same API as buttons.

Comment (torounit) [reply packages/interface/src/components/complementary-area-toggle/index.js:38]: it's better that way. I'll do it.

Comment (torounit) [reply packages/interface/src/components/complementary-area-toggle/index.js:38]: @youknowriad fixed 👍

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by gziolo** [comment]
Fixes #21430

**Pull request merged**